### ISSUE TITLE
Alerts Page Error

### ISSUE THREAD

**Issue by SychO9** [SimpleMachines/SMF#5156]
#### Description
```
Type of error: Undefined
8: Undefined index: board
http://localhost/SMF21/index.php?action=profile;area=showalerts
C:/Apache24/htdocs/SMF21/Sources/Profile-Modify.php (Line 2325)
````
https://github.com/SimpleMachines/SMF2.1/blob/54e2dbcce94b0832b9cae7a9aa1494b19bea9279/Sources/Profile-Modify.php#L2325-L2326

I guess this appeared after the changes here #4834 

### Steps to reproduce
1. Produce an alert not post related, like a buddy request
2. Go to the other account, click Alerts (you see the alert but it isn't counted)
3. Click All Alerts, Check your error log

### Environment (complete as necessary)
- Version/Git revision: latest
- Database Type: MySQL
- Database Version: 5.7.23
- PHP Version: 7.1

**Comment by jdarwood007** [comment]
I haven't had time to test this but, does this resolve it?

Profile-Modify.php
```
-		if (!isset($boards[$extra['board']]))
-			unset($alerts[$id_alert]);		
+		if (isset($extra['board']) && !!isset($boards[$extra['board']]))
+			unset($alerts[$id_alert]);
```

**Comment by SychO9** [comment]
Well, it fixes the error, but the alerts system is still very bugged, alerts related to posts don't work at all, and I get duplicate alerts, also the alert count isn't working

Perhaps a new issue about it 'd be better.

**Comment by smfbigguy** [comment]
I was going to open an issue as per this thread:
https://www.simplemachines.org/community/index.php?topic=563540.0

but saw things were being worked on.

**Comment by smfbigguy** [comment]
I should say. I implemented the code change from jdarwood and it made no difference. I was not getting errors for this before that, but thought it would fix alerts somewhere else in the code for me. I still run this change. (third post up)

**Comment by SychO9** [comment]
I'm closing this as it has been fixed here #5193
